images/Containerfile.core.base

```diff
@@ -12,74 +16,83 @@ ENV PULP_SETTINGS=/etc/pulp/settings.py
-#
 # The last 5 lines (before clean) are needed until python3-createrepo_c gets an
-# RPM upgrade to 0.15.10. Until then, we install & build it from PyPI.
-RUN		dnf -y update && \
-		dnf -y install wget git && \
-		dnf -y install libxcrypt-compat && \
-		dnf -y install libpq-devel && \
-		dnf -y install python3-psycopg2 && \
-		dnf -y install glibc-langpack-en && \
-		dnf -y install python3-libmodulemd && \
-		dnf -y install openldap-devel && \
-		dnf -y install skopeo && \
-		dnf -y install openssl && \
-		dnf -y install openssl-devel && \
-		dnf -y install ostree && \
-		dnf -y install ostree-libs && \
-		dnf -y install python3-devel && \
-		dnf -y install python3-setuptools && \
-		dnf -y install python-pip && \
-		dnf -y install gcc && \
-		dnf -y install libmodulemd-devel && \
-		dnf -y install libcomps-devel && \
-		dnf -y install ninja-build && \
-		dnf -y install 'dnf-command(builddep)' && \
-		dnf -y install cairo-devel && \
-		dnf -y install cmake && \
-		dnf -y install gobject-introspection-devel && \
-		dnf -y install cairo-gobject-devel && \
-		dnf -y builddep createrepo_c && \
-		dnf clean all
+# RPM upgrade to 0.16.2. Until then, we install & build it from PyPI.
+#
+# TODO: Investigate differences between `dnf builddep createrepo_c` vs the list
+# of dependencies below. For example, drpm-devel.
+RUN dnf -y install dnf-plugins-core && \
+    dnf -y install https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm && \
+    dnf config-manager --set-enabled powertools && \
+    dnf -y update && \
+    dnf -y install python38 python38-devel && \
+    dnf -y install openssl && \
+    dnf -y install openldap-devel && \
+    dnf -y install wget git && \
+    dnf -y install python3-psycopg2 && \
+    dnf -y install glibc-langpack-en && \
+    dnf -y install python3-libmodulemd && \
+    dnf -y install python3-libcomps && \
+    dnf -y install libpq-devel && \
+    dnf -y install python3-setuptools && \
+    dnf -y install buildah --exclude container-selinux && \
+    dnf -y install xz && \
+    dnf -y install libmodulemd-devel && \
+    dnf -y install libcomps-devel && \
+    dnf -y install zchunk-devel && \
+    dnf -y install ninja-build && \
+    dnf -y install cairo-devel cmake gobject-introspection-devel cairo-gobject-devel && \
+    dnf -y install libcurl-devel libxml2-devel sqlite-devel file-devel && \
+    dnf -y install ostree-libs ostree && \
+    dnf -y install skopeo && \
+    dnf -y install podman && \
+    dnf -y install sudo && \
+    dnf -y install zstd && \
+    dnf clean all
+
+# Needed to prevent the wrong version of cryptography from being installed,
+# which would break PyOpenSSL.
+RUN pip3 install --upgrade pip && \
+    rm -rf /root/.cache/pip
 
 # Need to install optional dep, rhsm, for pulp-certguard
-RUN pip install  \
-	rhsm \
-	setproctitle \
-	gunicorn \
-        python-nginx \
-	"django-storages[boto3,azure]>=1.12.2" \
-	"requests[use_chardet_on_py3]" \
-	pulpcore
```

Comment:
I would keep pulpcore, so the following images would just update some dependencies, it wouldn't need to download everything

Reply:
I disagree.

For one thing, we may end up downgrading pulpcore later on. Because some container files let you specify the version of pulpcore.

2nd, it's a clean break between dependencies vs application.